libs/domain/cart/src/controllers/cart.controller.ts

```diff
@@ -21,22 +23,52 @@ export class CartController {
     this.observe = new ObserveController(host);
   }
 
+  protected get cartQualifier(): Observable<CartQualifier | undefined> {
+    return this.observe
+      .get('cartId')
+      .pipe(switchMap((cartId) => of(cartId ? { cartId } : undefined)));
```

Comment:
```suggestion
      .pipe(switchMap((cartId) => of(cartId ? { cartId } : undefined)));
```
Why do we need this part?